root/static/scripts/edit/URLCleanup.js

```diff
@@ -1223,6 +1223,26 @@ const CLEANUPS = {
       return url;
     }
   },
+  niconicovideo: {
+    match: [new RegExp("^(https?://)?([^/]+\\.)?(nicovideo\\.jp/)","i")],
+    type: _.default({}, LINK_TYPES.streamingmusic, LINK_TYPES.videochannel),
+    clean: function(url) {
+      var m = /^(?:https?:\/\/)?(?:[^\/]+\.)?nicovideo\.jp\/(user\/[0-9]+|watch\/sm[0-9]+).*$/, "https://www.nicovideo.jp/$1";
+    }
+    validate: function (url, id) {
+      var m = /^(?:https?:\/\/)?(?:[^\/]+\.)?nicovideo\.jp\/(user\/[0-9]+|watch\/sm[0-9]+).*$/.exec(url);
+      if (m){
+        var prefix = m[1];
+        switch (id){
+          case LINK_TYPES.streamingmusic.recording || LINK_TYPES.streamingmusic.release:
+            return prefix === 'watch\/sm[0-9]+'
```

Comment:
This comparison won't work because so-called prefix will be something like `watch/sm123578`.
If you apply above suggestion, you can simply return `prefix === 'watch'` instead.
Final semi-colon `;` is missing too.